runtime/runtime/src/verifier.rs

```diff
@@ -96,16 +110,42 @@ pub fn validate_transaction(
     signed_tx: SignedTransaction,
     current_protocol_version: ProtocolVersion,
 ) -> Result<ValidatedTransaction, (InvalidTxError, SignedTransaction)> {
-    if let Err(err) = validate_actions(
-        &config.wasm_config.limit_config,
-        signed_tx.transaction.actions(),
-        current_protocol_version,
-    ) {
-        return Err((InvalidTxError::ActionsValidation(err), signed_tx));
+    if let Err(err) = validate_transaction_actions(&config, &signed_tx, current_protocol_version) {
+        return Err((err, signed_tx));
     }
     ValidatedTransaction::new(config, signed_tx)
 }
 
+/// Validates a transaction contains well-formed actions and is valid for the given runtime config.
+///
+/// This function is similar to `validate_transaction` but does NOT verify the signature.
+pub(crate) fn validate_transaction_well_formed<'a>(
+    config: &RuntimeConfig,
+    signed_tx: &SignedTransaction,
+    current_protocol_version: ProtocolVersion,
+) -> Result<(), InvalidTxError> {
+    validate_transaction_actions(config, signed_tx, current_protocol_version)?;
+    ValidatedTransaction::check_valid_for_config(config, signed_tx)
+}
+
+/// Returns the signature and public key if they are of ED25519 type.
+///
+/// Used for batch signature verification, which only supports ED25519 signatures.
+/// Returns `None` if the signature or public key are not ED25519.
+pub(crate) fn get_batchable_signature_and_public_key(
```

Comment:
Nit: this seems out of place in this module now. The module otherwise has nothing to do with batched signature validation. Maybe move to `lib.rs`?